homeassistant/components/home_plus_control/config_flow.py

```diff
@@ -0,0 +1,161 @@
+        except Exception as err:  # pylint: disable=broad-except
+            self.logger.warning("Error retrieving authentication token: %s", err)
+            return self.async_abort(reason="oauth_error")
+
+        token["expires_at"] = time.time() + token["expires_in"]
+        self.logger.info("Successfully authenticated")
+
+        config_data = {}
+        config_data[CONF_CLIENT_ID] = self.flow_impl.client_id
+        config_data[CONF_CLIENT_SECRET] = self.flow_impl.client_secret
+        config_data[CONF_SUBSCRIPTION_KEY] = self.flow_impl.subscription_key
+
+        return self.async_create_entry(
+            title=self.flow_impl.name,
+            data={
+                "auth_implementation": self.flow_impl.domain,
+                "token": token,
+                **config_data,
+            },
+        )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Return the Options Flow Handler for the Home+ Control component."""
+        return HomePlusControlOptionsFlowHandler(config_entry)
+
+
+class HomePlusControlOptionsFlowHandler(config_entries.OptionsFlow):
+    """Options flow handler for the Home+ Control integration."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry):
+        """Initialize the HomePlusControlOptionsFlowHandler object."""
+        self.config_entry = config_entry
+        self.options = dict(config_entry.options)
+
+        self.options.setdefault(
+            CONF_PLANT_UPDATE_INTERVAL,
+            DEFAULT_UPDATE_INTERVALS[CONF_PLANT_UPDATE_INTERVAL],
+        )
+        self.options.setdefault(
+            CONF_PLANT_TOPOLOGY_UPDATE_INTERVAL,
+            DEFAULT_UPDATE_INTERVALS[CONF_PLANT_TOPOLOGY_UPDATE_INTERVAL],
+        )
+        self.options.setdefault(
+            CONF_MODULE_STATUS_UPDATE_INTERVAL,
+            DEFAULT_UPDATE_INTERVALS[CONF_MODULE_STATUS_UPDATE_INTERVAL],
+        )
+
+    async def async_step_init(self, user_input=None):
+        """Handle the integration options."""
+        assert self.hass
+        errors = {}
+
+        if user_input is not None:
+            # Validate user input
+            valid = await self._is_valid(user_input, errors)
+            if valid:
+                return self.async_create_entry(title="Home+ Control", data=user_input)
+
+        options_schema = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_PLANT_UPDATE_INTERVAL,
+                    default=self.options[CONF_PLANT_UPDATE_INTERVAL],
+                ): cv.string,
+                vol.Optional(
+                    CONF_PLANT_TOPOLOGY_UPDATE_INTERVAL,
+                    default=self.options[CONF_PLANT_TOPOLOGY_UPDATE_INTERVAL],
+                ): cv.string,
+                vol.Optional(
+                    CONF_MODULE_STATUS_UPDATE_INTERVAL,
+                    default=self.options[CONF_MODULE_STATUS_UPDATE_INTERVAL],
+                ): cv.string,
```

Comment:
The problem that I had is that the API offered by Legrand has a very limited quota of requests - only 500 per day. In my initial tests, where I was refreshing data every minute, I soon ran into problems because it was easy to reach the limit and started getting HTTP 403 responses until the next day.

By including then these 3 refresh intervals I was hoping to give the more advanced users the possibility of deciding how often they want to get data updates (at their own risk of reaching the limit!). I am assuming that the plant information and the plant topology won't change that often once a user has setup the modules in their home and that the more common update that will be required will be simply updating the status of the modules themselves. 

However, I also understand that there might be some users that could want to update plant information and topology more frequently at specific times, eg. when making changes in the layout or adding/removing modules.

For this reason, I thought that having these 3 refresh intervals was a reasonable option. The selected default values are aligned with my assumption that plant topology will change less frequently than module status.

Reply:
I suggest we set a sane default for now and remove the options. We want to avoid users misconfiguring themselves into rate limits.

Reply:
OK, I am removing these options then from the component along with the options flow.